### ISSUE TITLE
VMNull when trying to get CLIENT::.keys

### ISSUE THREAD

**Issue by AlexDaniel** [rakudo/rakudo#2775]
```
say CLIENT::.keys
```

```
¦«2016.01.1,2016.02,2016.03,2016.04,2016.05,2016.06,2016.07.1,2016.08.1,
  2016.09,2016.10,2016.11,2016.12,2017.01,2017.02,2017.03,2017.04.3,
  2017.05,2017.06,2017.07,2017.08,2017.09,2017.10,2017.11,2017.12,2018.01,
  2018.02.1,2018.03,2018.04.1,2018.05,2018.06,2018.08,2018.09,2018.10,
  2018.11,2018.12»:
GLOBAL can have no client package
  in block <unit> at /tmp/TbnvAyjGet line 1

 «exit code = 1»
¦«2019.03.1,HEAD(4825f31)»:
Cannot invoke this object (REPR: Null; VMNull)
  in block <unit> at /tmp/TbnvAyjGet line 1

«exit code = 1»
```

Bisected: (2018-12-26) https://github.com/rakudo/rakudo/commit/5540212cfc8805f869ce566c94c406f5cb18224b

**Comment by lizmat** [comment]
> On 20 Mar 2019, at 11:44, Aleks-Daniel Jakimenko-Aleksejev <notifications@github.com> wrote:
> 
> say CLIENT::.keys
> 
> ¦«2016.01.1,2016.02,2016.03,2016.04,2016.05,2016.06,2016.07.1,2016.08.1,
>   2016.09,2016.10,2016.11,2016.12,2017.01,2017.02,2017.03,2017.04.3,
>   2017.05,2017.06,2017.07,2017.08,2017.09,2017.10,2017.11,2017.12,2018.01,
>   2018.02.1,2018.03,2018.04.1,2018.05,2018.06,2018.08,2018.09,2018.10,
>   2018.11,2018.12»:
> GLOBAL can have no client package
>   in block <unit> at /tmp/TbnvAyjGet line 1
> 
>  «exit code = 1»
> ¦«2019.03.1,HEAD(4825f31)»:
> Cannot invoke this object (REPR: Null; VMNull)
>   in block <unit> at /tmp/TbnvAyjGet line 1
> 
> «exit code = 1»
> 
> Bisected: (2018-12-26) 5540212

This one is **WEIRD**.  I’ve added the following debug code, the line number in the stack trace is the line with “die”.

nqp::say($cur.^name);
            my $pkg := nqp::getlexrel(
                nqp::getattr(nqp::decont($cur), PseudoStash, '$!ctx'),
                '$?PACKAGE');
nqp::say($pkg.^name);
nqp::say($pkg.^name.^name);
            die "GLOBAL can have no client package" if $pkg.^name eq "GLOBAL”;

Then this gives:

$ perl6 -e ‘CLIENT::'
PseudoStash
GLOBAL
Str
Cannot invoke this object (REPR: Null; VMNull)
   at SETTING::src/core/PseudoStash.pm6:175  (...install/share/perl6/runtime/CORE.setting.moarvm:)

Looks like it’s dying in the eq ??


And the problem goes away when removing that part https://github.com/rakudo/rakudo/commit/5540212cfc8805f869ce566c94c406f5cb18224b .   *sigh*

**Comment by lizmat** [comment]
Fixed with 6599998c2c .  Unsure how we should test for this.
